department-of-veterans-affairs/va.gov-cms
# VACMS-4850 Vet Center dashboard MVP

## Description
See #4850

## Testing done
Visual

## Screenshots
![image](https://user-images.githubusercontent.com/2404547/114211793-12a13700-992f-11eb-8687-dbe648c9bcff.png)
![image](https://user-images.githubusercontent.com/2404547/114211807-1634be00-992f-11eb-9657-cea3fae6150a.png)
![image](https://user-images.githubusercontent.com/2404547/114211817-19c84500-992f-11eb-93d2-d6d6b403f13b.png)

## QA steps
As an administrator, go to `/taxonomy/term/190/edit` and choose the product: `Vet Centers`.
- Clear cache, and go to `/section/veterans-health-administration/vet-centers/escanaba-vet-center`
- [ ] Visually verify that you see the new dashboard format (6 cards up top, listing down below)
![image](https://user-images.githubusercontent.com/5752113/116339697-e3eae380-a7ab-11eb-88c1-0cd35f424a45.png)

- [ ]  verify view and edit links on the 6 cards work as expected.

- Go to /section/veterans-health-administration/daytona-beach-vet-center
- [ ] verify that only 5 cards appear as there is no locations page created yet for daytona. (fails gracefully)
![image](https://user-images.githubusercontent.com/5752113/116339987-65427600-a7ac-11eb-9389-7690bf072b89.png)


-  Go to a vetcenter section that has no matching node like /section/veterans-health-administration/vet-centers-mvp
![image](https://user-images.githubusercontent.com/5752113/116340774-bb63e900-a7ad-11eb-94c8-c9777b88d6be.png)


- [ ] validate that no cards appear and there are no errors  on the page (fails gracefully)
- [ ] Visit a health system section `/section/veterans-health-administration/vamc-facilities/va-salisbury-health-care` and verify that there is no dashboard and it functions as expected 
 ![image](https://user-images.githubusercontent.com/5752113/116340582-67590480-a7ad-11eb-8c27-f20015b4afbd.png)
![image](https://user-images.githubusercontent.com/5752113/116340582-67590480-a7ad-11eb-8c27-f20015b4afbd.png)Visit another administration term page, and visually verify that the default view listing is displayed, e.g.: `/section/veterans-affairs`

CACHING Testing with xdebug:
- [x] Validate that the service is only called once during the initial sectionpage load
- [x] Validate that the block build() function is only called on the first page load.
- [x] Validate subsequent lods of the page do not run the service or the build(). (cached indefinitely)
- [x] Validate cache is only invalidated by cache flush OR editing he section and saving the section.


## Release instructions
- [ ] As an administrator, go to `/taxonomy/term/190/edit` and choose the product: `Vet Centers`.

### Definition of Done


### Select Team for PR review


### Is this PR blocked by another PR?


### Does this PR need review from a Product Owner


### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
